### PR TITLE
Housekeeping 

### LINKED ISSUE #107: rename `distance_to_f` and `predictorID` to `distance_to_forager` and `predictor_name`

### PR DESCRIPTION
**Pull request by PalkaPuri**
Unexpectedly large PR (sorry) with some housekeeping tasks 
[Note : this includes changes from pp-collab2-derivepredictors, so the number of changes would drastically reduce if that PR is merged first!]

- Added docstrings to local_windows.py  
- Renamed `predictorID` to `predictor_name`, and `distance_to_f` to `distance_to_forager` -- resolves #107 
- Updating passing of kwargs to constraint functions [modules updated: filtering.py, local_windows.py, velocity.py, proximity.py(WIP)]   
- Removed unused argument `fps` from `subsample_frames_constant_frame_rate`
- Added docstrings and updated warnings in `dataObject`
- Updated docstring formatting in velocity.py
- Added collab2 notebooks to automatic testing pipeline + made sure the notebooks are running in light of these tweaks (WIP for proximity_predictor.ipynb)

### CODE REVIEW

File: collab2/foraging/toolkit/filtering.py [hunk 1]
```diff
@@ -1,31 +1,19 @@
...
-
-# TODO keep in mind and test for this functionality in the future
-
 
 def filter_by_distance(
     foragersDF: pd.DataFrame,
     f: int,
     t: int,
     interaction_length: float,
     interaction_constraint: Optional[
```
Comment (PalkaPuri) [inline collab2/foraging/toolkit/filtering.py:13]: @rfl-urbaniak, including `...` in type hints for `interaction_constraint` gives an error here

File: collab2/foraging/toolkit/velocity.py [hunk 2]
```diff
@@ -85,16 +80,16 @@ def _generic_velocity_predictor(
...
     foragersDF: pd.DataFrame,
     local_windows: List[List[pd.DataFrame]],
-    predictorID: str,
+    predictor_name: str,
     interaction_length: float,
     dt: int,
     sigma_v: float,
     sigma_t: float,
     transformation_function: Callable[[pd.DataFrame], pd.DataFrame],
     interaction_constraint: Optional[
```
Comment (PalkaPuri) [inline collab2/foraging/toolkit/velocity.py:89]: @rfl-urbaniak this is another instance of `...` in type hints error

File: collab2/foraging/toolkit/local_windows.py [hunk 3]
```diff
@@ -34,30 +49,45 @@ def get_grid(
...
-    grid_constraint: Optional[
-        Callable[[pd.DataFrame, pd.DataFrame, Any], pd.DataFrame]
-    ] = None,
-    grid_constraint_params: Optional[dict] = None,
-) -> pd.DataFrame:
-
-    # Note: args `grid_size`, `num_foragers`, `num_frames` are not exposed to users but set to
-    # values inherited from `foragers_object`` by `generate_local_windows`.
+    grid_constraint: Optional[Callable[[pd.DataFrame], pd.DataFrame]] = None,
+    **grid_constraint_params,
```
Comment (PalkaPuri) [inline collab2/foraging/toolkit/local_windows.py:57]: @rfl-urbaniak this is another instance of ... in type hints error

File: collab2/foraging/toolkit/local_windows.py [hunk 4]
```diff
@@ -5,23 +5,38 @@
...
+    grid_size: int,
     sampling_fraction: float = 1.0,
-    random_seed: int = 0,
-    grid_constraint: Optional[
-        Callable[[pd.DataFrame, pd.DataFrame, Optional[dict]], pd.DataFrame]
-    ] = None,
-    grid_constraint_params: Optional[dict] = None,
+    random_seed: Optional[int] = 0,
+    grid_constraint: Optional[Callable[[pd.DataFrame], pd.DataFrame]] = None,
+    **grid_constraint_params,
```
Comment (PalkaPuri) [inline collab2/foraging/toolkit/local_windows.py:16]: @rfl-urbaniak this is another instance of ... in type hints error

**Review by rfl-urbaniak — APPROVED** [review]

**Pull request merged**